src/main/webapp/data-visualization.js

```diff
@@ -129,21 +128,28 @@ function createDataArray(censusDataArray, isCountyQuery) {
   const vizDataArray = [];
   // first row is headers
   const regionIndex = censusDataArray[0].indexOf('state');
-  censusDataArray = censusDataArray.splice(1); // get rid of header row
+  censusDataArray = censusDataArray.slice(1); // get rid of header row
   // Check to see if an extra calculation for percentages is needed
+  const table = initDataTable(isCountyQuery);
```

Comment:
Instead of creating a table element here start collecting the data in variable(s) and then just before the 
"document.getElementById('data-table').appendChild(table);" line you can call one function that creates the table. This isolates all the table creation stuff to one function and you easily replace that with some other way to display the data should you choose to.